package/MDAnalysis/core/groups.py

```diff
@@ -37,84 +37,156 @@ def make_classes():
 
     Returns
     -------
-    A dictionary with a copy of all MDA container classes
+    Two dictionaries. One with a set of :class:`TopologyAttrContainer` classes 
+    to serve as bases for universe-specific MDA container classes. Another,
+    with the final merged versions of those classes. The classes themselves are
+    used as hashing keys.
     """
-    def copy_class(newname, *parents):
-        return type(newname, tuple(parents), {})
-
-    classdict = {}
-    # The 'GB' middle man is needed so that a single topologyattr
+    bases = {}
+    classes = {}
+    # The 'GBase' middle man is needed so that a single topologyattr
     #  patching applies automatically to all groups.
-    GB = classdict['group'] = copy_class(
-        'Group', GroupBase)
-    AG = classdict['atomgroup'] = copy_class(
-        'AtomGroupBase', GB)
-    RG = classdict['residuegroup'] = copy_class(
-        'ResidueGroupBase', GB)
-    SG = classdict['segmentgroup'] = copy_class(
-        'SegmentGroupBase', GB)
+    GBase = bases[GroupBase] = _make_attr_container()
+    AGBase = bases[AtomGroup] = _make_attr_container(base=GBase)
+    RGBase = bases[ResidueGroup] = _make_attr_container(base=GBase)
+    SGBase = bases[SegmentGroup] = _make_attr_container(base=GBase)
     # In the current Group-centered topology scheme no attributes apply only
     #  to ComponentBase, so no need to have a 'CB' middle man.
-    #CB = classdict['component'] = copy_class(
-    #    'ComponentBase', ComponentBase)
-    A = classdict['atom'] = copy_class(
-        'AtomBase', ComponentBase)
-    R = classdict['residue'] = copy_class(
-        'ResidueBase', ComponentBase)
-    S = classdict['segment'] = copy_class(
-        'SegmentBase', ComponentBase)
+    #CBase = _make_attr_container(singular=True)
+    ABase = bases[Atom] = _make_attr_container(singular=True)
+    RBase = bases[Residue] = _make_attr_container(singular=True)
+    SBase = bases[Segment] = _make_attr_container(singular=True)
+
+    # Initializes the class cache.
+    classes[AtomGroup] = AGBase._mix('AtomGroup', AtomGroup)
+    classes[ResidueGroup] = RGBase._mix('ResidueGroup', ResidueGroup)
+    classes[SegmentGroup] = SGBase._mix('SegmentGroup', SegmentGroup)
+
+    classes[Atom] = ABase._mix('Atom', Atom)
+    classes[Residue] = RBase._mix('Residue', Residue)
+    classes[Segment] = SBase._mix('Segment', Segment)
+
+    return bases, classes
+
+
+class TopologyAttrContainer(object):
```

Comment:
I guess that should be safe. In that case I'll also move the `_make_attr_container` into the class.